### DISCUSSION TITLE
Refresh helm repositories only once per run

### DISCUSSION THREAD

**Discussion by luispabon** [helmfile/helmfile#1306] [Ideas]
Currently, helmfile refreshes helm repositories for each chart that is matched by a selector (or no selector). This is not a huge issue with a low amount of repositories, but it's a problem on larger codebases.

A chart codebase I'm currently working on has 17 helm repositories referenced, and builds can take a huge amount of time as a result - not just deployments, where we might target a single chart in a selector or a small group of them, but verification steps on the repository build (eg rendering to template then running via kubeconform or whatever) take a huge amount of time, 90% of which is spent on helm refreshing the same 17 repositories over and over again, once per chart involved in the operation.

**Comment by yxxhero** [comment]
Good idea. Could you post some key  logs？

**Reply by kxkcx** [reply]
I am highly interested in this problem. Can it be assigned to me?

**Comment by luispabon** [comment]
Here's a log from a helmfile template operation targeting the entire repo, 63 charts in total. Just copy&pasted the dependency side of things. 12 minutes total. Some chart names anonimised.

https://pastebin.com/Ta1HJUem

You can see what's happening there.
 * First, it adds all helm repos found on the helmfiles
 * Second, it loops through each release and issues a dependency update, which triggers ALL helm repos to be updated
 * Third, install any chart dependencies for that release 

Ideally:
 * First, same
 * Second, update helm repos only once
 * Third, loop through each release and install chart dependencies

**Reply by yxxhero** [reply]
@luispabon got it. it will be an important feature.
